File: ticket-frontend/src/modules/fuzzy/fuzzy-variable-popularity.ts
import { FuzzyVariableDistributionPart, FuzzyVariableDistributionPartUtils, FuzzyVariableI } from './fuzzy-variable-distribution';

export class FuzzyVariablePopularity implements FuzzyVariableI {
    varVeryPopular: FuzzyVariableDistributionPart;
    varPopular: FuzzyVariableDistributionPart;
    varAverage: FuzzyVariableDistributionPart;
    varUnpopular: FuzzyVariableDistributionPart;

    constructor(obj: {
        varVeryPopular: FuzzyVariableDistributionPart,
        varPopular: FuzzyVariableDistributionPart,
        varAverage: FuzzyVariableDistributionPart,
        varUnpopular: FuzzyVariableDistributionPart
    }) {
        this.varVeryPopular = obj.varVeryPopular;
        this.varPopular = obj.varPopular;
        this.varAverage = obj.varAverage;
        this.varUnpopular = obj.varUnpopular;
    }

    static fromObj(obj: any): FuzzyVariablePopularity {
        return new FuzzyVariablePopularity({
            varVeryPopular: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varVeryPopular),
            varPopular: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varPopular),
            varAverage: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varAverage),
            varUnpopular: FuzzyVariableDistributionPartUtils.fuzzyVariableDistributionPartFromObj(obj.varUnpopular)
        })
    }

    getFuzzyVariableMap(): Record<string, FuzzyVariableDistributionPart> {
        return {
            [FuzzyVariablePopularityFields.VERY_POPULAR]: this.varVeryPopular,
            [FuzzyVariablePopularityFields.POPULAR]: this.varPopular,
            [FuzzyVariablePopularityFields.AVERAGE]: this.varAverage,
            [FuzzyVariablePopularityFields.UNPOPULAR]: this.varUnpopular,
        }
    }


    getFuzzyVariableColorsMap(): Record<string, string> {
        return {
            [FuzzyVariablePopularityFields.VERY_POPULAR]: 'green',
            [FuzzyVariablePopularityFields.POPULAR]: 'yellow',
            [FuzzyVariablePopularityFields.AVERAGE]: 'orange',
            [FuzzyVariablePopularityFields.UNPOPULAR]: 'red',
        }
    }

}

export enum FuzzyVariablePopularityFields {
    VERY_POPULAR = 'VERY_POPULAR',
    POPULAR = 'POPULAR',
    AVERAGE = 'AVERAGE',
    UNPOPULAR = 'UNPOPULAR',
}

File: ticket-backend/src/main/java/aics/domain/fuzzy/models/FuzzyVariablePopularity.java
package aics.domain.fuzzy.models;

import aics.domain.fuzzy.constants.FuzzyVariablePopularityFields;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.HashMap;

@Data
@Accessors(chain = true)
@AllArgsConstructor
public class FuzzyVariablePopularity implements Serializable, FuzzyVariableI {
    private FuzzyVariableDistributionPart varVeryPopular;
    private FuzzyVariableDistributionPart varPopular;
    private FuzzyVariableDistributionPart varAverage;
    private FuzzyVariableDistributionPart varUnpopular;

    @Override
    public HashMap<String, FuzzyVariableDistributionPart> getFuzzyVariableMap() {
        HashMap<String, FuzzyVariableDistributionPart> fuzzyVariableMap = new HashMap<>();
        fuzzyVariableMap.put(FuzzyVariablePopularityFields.VERY_POPULAR.name(), this.varVeryPopular);
        fuzzyVariableMap.put(FuzzyVariablePopularityFields.POPULAR.name(), this.varVeryPopular);
        fuzzyVariableMap.put(FuzzyVariablePopularityFields.AVERAGE.name(), this.varVeryPopular);
        fuzzyVariableMap.put(FuzzyVariablePopularityFields.UNPOPULAR.name(), this.varVeryPopular);
        return fuzzyVariableMap;
    }
}
